src/services/public_http_server/handlers/relay_webhook/handlers/notify_subscribe.rs

```diff
@@ -82,195 +91,215 @@ pub async fn handle(msg: RelayIncomingMessage, state: &AppState) -> Result<(), R
+                    .await
+                    .map_err(|e| RelayMessageServerError::NotifyServerError(e.into()))?; // TODO change to client error?
+            if let Some(welcome_notification) = welcome_notification {
+                info!("Welcome notification enabled");
+                if welcome_notification.enabled && scope.contains(&welcome_notification.r#type) {
+                    info!("Scope contains welcome notification type, sending welcome notification");
+                    let notification = upsert_notification(
+                        Uuid::new_v4().to_string(),
+                        project.id,
+                        Notification {
+                            r#type: welcome_notification.r#type,
+                            title: welcome_notification.title,
+                            body: welcome_notification.body,
+                            url: welcome_notification.url,
+                            icon: None,
+                        },
+                        &state.postgres,
+                        state.metrics.as_ref(),
+                    )
+                    .await
+                    .map_err(|e| RelayMessageServerError::NotifyServerError(e.into()))?; // TODO change to client error?
+
+                    upsert_subscriber_notifications(
+                        notification.id,
+                        &[subscriber.id],
+                        &state.postgres,
+                        state.metrics.as_ref(),
+                    )
+                    .await
+                    .map_err(|e| RelayMessageServerError::NotifyServerError(e.into()))?;
+                // TODO change to client error?
+                } else {
+                    info!("Scope does not contain welcome notification type, not sending welcome notification");
+                }
             } else {
-                info!("Scope does not contain welcome notification type, not sending welcome notification");
+                info!("Welcome notification not enabled");
             }
         } else {
-            info!("Welcome notification not enabled");
+            info!("Subscriber already existed, not sending welcome notification");
         }
-    } else {
-        info!("Subscriber already existed, not sending welcome notification");
-    }
 
-    // TODO do in same transaction as upsert_subscriber()
-    state
-        .notify_webhook(
-            project.project_id.as_ref(),
-            // TODO uncomment when `WebhookNotificationEvent::Updated` exists
-            // if subscriber.inserted {
-            WebhookNotificationEvent::Subscribed,
-            // } else {
-            // WebhookNotificationEvent::Updated
-            // },
-            account.as_ref(),
+        // TODO do in same transaction as upsert_subscriber()
+        state
+            .notify_webhook(
+                project.project_id.as_ref(),
+                // TODO uncomment when `WebhookNotificationEvent::Updated` exists
```

Comment:
I'm curious about this event, does it exists already?

Reply:
No it does not